Formula/netcdf-mpi.rb

```diff
@@ -0,0 +1,50 @@
+class NetcdfMpi < Formula
+  desc "The NetCDF C I/O library compiled with MPI parallel features enabled"
+  homepage "https://www.unidata.ucar.edu/software/netcdf/"
+  url "https://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-c-4.7.4.tar.gz"
+  sha256 "0e476f00aeed95af8771ff2727b7a15b2de353fb7bb3074a0d340b55c2bd4ea8"
+  license "BSD-3-Clause"
+
+  depends_on "hdf5-mpi"
+
+  conflicts_with "netcdf", :because => "netcdf-mpi is an MPI parallel build, use either netcdf or netcdf-mpi"
```

Comment:
```suggestion
  keg_only "netcdf-mpi is an MPI parallel build of netcdf"
```
With `conflicts_with "netcdf"`, you won't be able to install anything that depends on `netcdf`. Instead, make this formula `keg_only`.

Reply:
`keg_only` is not what we want - this install should be linked into `/usr/local` so that it's easily found and used.

formula that depend on `netcdf`'s install of the netcdf-c library can be updated to depend on either `netcdf` OR `netcdf-mpi`. Those codes will compile, link and run without any modification.

Reply:
Homebrew unfortunately has no mechanism to support alternate dependencies. There is no way to depend on `netcdf` OR `netcdf-mpi`.